### PR TITLE
VS-759 (and VS-760)

### PR DESCRIPTION
**Pull request by gbggrant**
Add more testing for create_genes_bqloadjson_from_annotations and create_vt_bqloadjson_from_annotations

### CODE REVIEW

File: scripts/variantstore/wdl/extract/test_create_vt_bqloadjson_from_annotations.py [hunk 1]
```diff
@@ -1,17 +1,39 @@
 import json
 import unittest
 import copy
+import gzip
+import tempfile
 
 from create_vt_bqloadjson_from_annotations import make_annotated_json_row
+from create_vt_bqloadjson_from_annotations import make_annotation_json
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/test_create_vt_bqloadjson_from_annotations.py:8]:
```suggestion
from create_vt_bqloadjson_from_annotations import make_annotation_json, make_annotated_json_row
```

**Review by mcovarr — COMMENTED** [review]
OOC why are the expected files checked in compressed?

File: scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import tempfile
+
+from create_genes_bqloadjson_from_annotations import make_annotation_json
+
+dir='variant_annotation_table_test_files/'
+normal_annotated_genes_json=dir + "test_annotated_genes.json.gz"
+expected_output_annotated_genes_bq_json=dir + "expected_test_output_annotated_genes_bq.json.gz"
+
+# An old style (before we separated positions and genes jsons into separate files) - expected to fail
+old_annotated_genes_json=dir + "test_old_annotated_genes.json.gz"
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py:12]: could bring this into the test method as it's not referenced anywhere else

**Comment by gbggrant** [comment]
@mcovarr I checked the test files in compressed as I was trying to save space, but I kind of agree that it's not very necessary (and they're not that big), so will decompress.

**Review by RoriCremer — APPROVED** [review]

File: scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import unittest
+import gzip
+import tempfile
+
+from create_genes_bqloadjson_from_annotations import make_annotation_json
+
+dir='variant_annotation_table_test_files/'
+normal_annotated_genes_json=dir + "test_annotated_genes.json.gz"
+expected_output_annotated_genes_bq_json=dir + "expected_test_output_annotated_genes_bq.json.gz"
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py:9]: Yup, fixed that.

File: scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py (hunk 2 above)
Comment (gbggrant) [reply scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py:12]: Yup, fixed that.

File: scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+class TestCreateGenesBqloadjsonFromAnnotations(unittest.TestCase):
+    def test_normal(self):
+        output_json = tempfile.NamedTemporaryFile(prefix="test_output_annotated_genes_bq", suffix=".json.gz")
+        make_annotation_json(normal_annotated_genes_json, output_json.name)
+        contents = gzip.open(output_json, 'r').read()
+        expected_contents = gzip.open(expected_output_annotated_genes_bq_json, 'r').read()
+        self.assertEqual(contents, expected_contents)
+
+    # This test tries to read an old-style json file and verifies that the program will fail if fed such a file
+    @unittest.expectedFailure
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply scripts/variantstore/wdl/extract/test_create_genes_bqloadjson_from_annotations.py]: I've updated the code to capture stderr and verify that the failure is for the expected reason.

**Pull request merged**